### PR TITLE
Fixed #29664 --Caught TypeError exception and giving out meaning full…

### PR DESCRIPTION
**Pull request by emiamar**
Fixed #29664 --Caught TypeError exception and giving out meaning full error message

### CODE REVIEW

File: django/core/handlers/base.py [hunk 1]
```diff
@@ -121,7 +121,13 @@ def _get_response(self, request):
         if response is None:
             wrapped_callback = self.make_view_atomic(callback)
             try:
-                response = wrapped_callback(request, *callback_args, **callback_kwargs)
+                response = wrapped_callback(
+                    request, *callback_args, **callback_kwargs)
+            except TypeError:
```
Comment (jeffyancey) [inline django/core/handlers/base.py]: What if something else throws this `TypeError`? If we want to give a message like this at all, I think we need to find a better way of ensuring the case you mentioned is causing it.

**Comment by emiamar** [comment]
I agree. This error is actually caused because view class  itself is instantiated , when one forgets to call the method `as_view` method. Since can view class inherits base class `View` its `__init__ ` method takes only 2 args.

**Comment by charettes** [comment]
I think the best way to detect this would be in `View.__new__` or `__init__`.

**Comment by carltongibson** [comment]
I've marked this `wontfix` on Trac. https://code.djangoproject.com/ticket/29664

If anyone thinks there's a reasonable approach to take please re-open.

**Pull request closed without merging**